subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -241,6 +250,44 @@ Only secure protocols like `https` are accepted.
 This is required to make sure no one can tamper with the download in flight.
 ====
 
+[[sub:download_repositories]]
+=== Toolchain Download Repositories
+
+Starting with Gradle 7.6, there is a generic way to configure arbitrary repositories to download toolchains from.
+
+[NOTE]
+====
+This new mechanism has priority over the Adoptium/AdoptOpenJDK built-in default, if configured the default will no longer be used for downloading.
+In future versions, the built-in default will be completely removed.
+====
+
+Configuration toolchain download repositories starts by applying one or more settings plugins, which provide implementations for them.
```

Comment:
It would be helpful to link to such a plugin sample, or documentation about how to write such a plugin. Otherwise, the rest of the documentation feels odd: you declare a repository, but it doesn't show by which magic it works.

Reply:
Good point. I fully intended to write the documentation for plugin authors, but while waiting to figure out the final version of things, I slowly forgot about it. Will do, thanks.